src/Components/Web.JS/src/Platform/WebAssemblyStartOptions.ts

```diff
@@ -14,4 +14,4 @@ export interface WebAssemblyStartOptions {
 // This type doesn't have to align with anything in BootConfig.
 // Instead, this represents the public API through which certain aspects
 // of boot resource loading can be customized.
-export type WebAssemblyBootResourceType = 'assembly' | 'pdb' | 'dotnetjs' | 'dotnetwasm' | 'timezonedata';
+export type WebAssemblyBootResourceType = 'assembly' | 'pdb' | 'dotnetjs' | 'dotnetwasm' | 'dynamicAssembly' | 'timezonedata';
```

Comment:
Could you clarify why we're distinguishing `assembly` from `dynamicAssembly` as far as the `loadBootResource` contract is concerned?

I'm worried that this will give people more opportunities to get things wrong by not realising that both `assembly` and `dynamicAssembly` exist. Could we use `assembly` for both of these cases?

Reply:
Distinguishing between `assembly` and `dynamicAssembly` allows us to:

- Make sure that assemblies marked as lazy-loaded don't get loaded during the app's startup
- Make sure that when an assembly is lazily-loaded at runtime that it was marked as so in the project config

Reply:
The `WebAssemblyBootResourceType` is what we pass into the `loadBootResource` callback. It isn't the same thing as the keys in the blazor.boot.json file. I understand we need to distinguish which assemblies are to be loaded lazily in blazor.boot.json, but I don't think we have to distinguish them when invoking the `loadBootResource` callback.

Again, sorry if I'm misunderstanding!

Reply:
Ah I see what you mean! My original response to this was going to be pointing out that we pass `dynamicAssembly` as the resource type when we call `loadResources` to load the lazy-loaded assembly but we don't.

We can remove this change and be fine. Good catch!